src/SmartTransactionsController.ts

```diff
@@ -423,29 +512,31 @@ export default class SmartTransactionsController extends BaseController<
   // ! Ask backend API to accept list of uuids as params
   async fetchSmartTransactionsStatus(
     uuids: string[],
+    { networkClientId }: { networkClientId?: NetworkClientId } = {},
   ): Promise<SmartTransaction[]> {
-    const { chainId } = this.config;
-
     const params = new URLSearchParams({
       uuids: uuids.join(','),
     });
-
+    const chainId = this.getChainId({ networkClientId });
+    const ethQuery = this.getEthQuery({ networkClientId });
     const url = `${getAPIRequestURL(
       APIType.BATCH_STATUS,
       chainId,
     )}?${params.toString()}`;
 
     const data = await this.fetch(url);
-
     Object.entries(data).forEach(([uuid, stxStatus]) => {
-      this.updateSmartTransaction({
-        statusMetadata: stxStatus as SmartTransactionsStatus,
-        status: calculateStatus(stxStatus as SmartTransactionsStatus),
-        cancellable: isSmartTransactionCancellable(
-          stxStatus as SmartTransactionsStatus,
-        ),
-        uuid,
-      });
+      this.#updateSmartTransaction(
+        {
+          statusMetadata: stxStatus as SmartTransactionsStatus,
+          status: calculateStatus(stxStatus as SmartTransactionsStatus),
+          cancellable: isSmartTransactionCancellable(
+            stxStatus as SmartTransactionsStatus,
+          ),
```

Comment:
```suggestion
          statusMetadata: stxStatus,
          status: calculateStatus(stxStatus),
          cancellable: isSmartTransactionCancellable(stxStatus),
```

Reply:
Follows from https://github.com/MetaMask/smart-transactions-controller/pull/237/files#r1402633206, which lets `stxStatus` be inferred.

Reply:
done here: https://github.com/MetaMask/smart-transactions-controller/pull/237/commits/93a05f14cbf91a344332dcf07d39f6924d5eacd3